mlir/include/Gradient/IR/GradientOps.td

```diff
@@ -119,6 +120,73 @@ def BackpropOp : Gradient_Op<"backprop", [AttrSizedOperandSegments]> {
 }
 
 
+def JVPOp : Gradient_Op<"jvp", [
+        SameVariadicOperandSize,
+        SameVariadicResultSize,
+        DeclareOpInterfaceMethods<CallOpInterface>,
+        DeclareOpInterfaceMethods<SymbolUserOpInterface>
+        ]> {
+    let summary = "Compute the jvp of a function.";
 
+    let arguments = (ins
+        StrAttr:$method,
+        FlatSymbolRefAttr:$callee,
+        Variadic<AnyType>:$params,
+        Variadic<AnyType>:$tangents,
+        OptionalAttr<AnyIntElementsAttr>:$diffArgIndices,
+        OptionalAttr<Builtin_FloatAttr>:$finiteDiffParam
+    );
+
+    let results = (outs
+        Variadic<AnyTypeOf<[AnyFloat, RankedTensorOf<[AnyFloat]>]>>:$calleeResults,
+        Variadic<AnyTypeOf<[AnyFloat, RankedTensorOf<[AnyFloat]>]>>:$jvps
+    );
+
+    let assemblyFormat = [{
+        $method $callee `(` $params `:` type($params) `,` $tangents `:` type($tangents) `)`
+        attr-dict `:` type($calleeResults) `,` type($jvps)
```

Comment:
Regarding my previous comment, the custom assembly still looks like the tangent arguments are being "passed" to the callee.
I think something like this extra `tangents` keyword works well in this situation to separate the function arguments from the tangent vectors.
We should also maintain a similar style to the grad op, which doesn't specify the callee argument types inline but rather at the end.
```suggestion
        $method $callee `(` $params `)` `tangents` `(` $tangents `)`
        attr-dict `:` functional-type(operands, results)
```